src/zenml/integrations/dash/visualizers/pipeline_run_lineage_visualizer.py

```diff
@@ -0,0 +1,344 @@
+from dash.dependencies import Input, Output
+
+from zenml.enums import ExecutionStatus
+from zenml.logger import get_logger
+from zenml.post_execution.pipeline_run import PipelineRunView
+from zenml.visualizers.base_pipeline_run_visualizer import (
+    BasePipelineRunVisualizer,
+)
+
+logger = get_logger(__name__)
+
+OVERALL_STYLE = {"fontFamily": "sans-serif"}
+COLOR_RED = "#AD5D4E"
+COLOR_BLUE = "#22577A"
+COLOR_YELLOW = "#FFB100"
+COLOR_GREEN = "#BFD7B5"
+
+STYLESHEET = [
+    # Group selectors
+    {
+        "selector": "node",
+        "style": {
+            "content": "data(label)",
+            "width": "70%",
+            "height": "70%",
+            "font-size": "24px",
+        },
+    },
+    {
+        "selector": "edge",
+        "style": {
+            "curve-style": "bezier",
+            "content": "data(label)",
+        },
+    },
+    # Class selectors
+    {
+        "selector": ".red",
+        "style": {
+            "background-color": COLOR_RED,
+            "line-color": COLOR_RED,
+            "target-arrow-color": COLOR_RED,
+        },
+    },
+    {
+        "selector": ".blue",
+        "style": {
+            "background-color": COLOR_BLUE,
+            "line-color": COLOR_BLUE,
+            "target-arrow-color": COLOR_BLUE,
+        },
+    },
+    {
+        "selector": ".yellow",
+        "style": {
+            "background-color": COLOR_YELLOW,
+            "line-color": COLOR_YELLOW,
+            "target-arrow-color": COLOR_YELLOW,
+        },
+    },
+    {
+        "selector": ".green",
+        "style": {
+            "background-color": COLOR_GREEN,
+            "line-color": COLOR_GREEN,
+            "target-arrow-color": COLOR_GREEN,
+        },
+    },
+    {"selector": ".rectangle", "style": {"shape": "rectangle"}},
+    {
+        "selector": ".edge-arrow",
+        "style": {"target-arrow-shape": "triangle"},
+    },
+    {"selector": ".solid", "style": {"line-style": "solid"}},
+    {"selector": ".dashed", "style": {"line-style": "dashed"}},
+]
+
+
+class PipelineRunLineageVisualizer(BasePipelineRunVisualizer):
+    """Implementation of a lineage diagram via the [dash](
+    https://plotly.com/dash/) and [dash-cyctoscape](
+    https://dash.plotly.com/cytoscape) library."""
+
+    ARTIFACT_PREFIX = "artifact_"
+    STEP_PREFIX = "step_"
+    STATUS_CLASS_MAPPING = {
+        ExecutionStatus.CACHED: "green",
+        ExecutionStatus.FAILED: "red",
+        ExecutionStatus.RUNNING: "yellow",
+        ExecutionStatus.COMPLETED: "blue",
+    }
+
+    @abstractmethod
+    def visualize(
+        self, object: PipelineRunView, *args: Any, **kwargs: Any
+    ) -> dash.Dash:
+        """Method to visualize pipeline runs via the Dash library. The layout
+        puts every layer of the dag in a column.
+        """
+
+        app = dash.Dash(
+            __name__,
+            external_stylesheets=[
+                dbc.themes.BOOTSTRAP,
+                dbc.icons.FONT_AWESOME,
```

Comment:
Correct me if I'm wrong, I couldn't find any use of FONT_AWESOME icons, maybe this could be removed here?
```suggestion
```

Reply:
Fixed in the latest commits